packages/app-store/routing-forms/lib/crmRouting/routerGetCrmContactOwnerEmail.ts

```diff
@@ -0,0 +1,78 @@
+import type { z } from "zod";
+
+import { getCRMContactOwnerForRRLeadSkip } from "@calcom/app-store/_utils/CRMRoundRobinSkip";
+import { EventTypeRepository } from "@calcom/lib/server/repository/eventType";
+import { SchedulingType } from "@calcom/prisma/enums";
+import type { ZResponseInputSchema } from "@calcom/trpc/server/routers/viewer/routing-forms/response.schema";
+
+import type { LocalRoute } from "../../types/types";
+import { enabledAppSlugs } from "../enabledApps";
+
+export default async function routerGetCrmContactOwnerEmail({
+  attributeRoutingConfig,
+  response,
+  action,
+}: {
+  attributeRoutingConfig: LocalRoute["attributeRoutingConfig"];
+  response: z.infer<typeof ZResponseInputSchema>["response"];
+  action: LocalRoute["action"];
+}) {
+  // Check if route is skipping CRM contact check
+  if (attributeRoutingConfig?.skipContactOwner) return null;
+
+  // Check if email is present
+  let prospectEmail: string | null = null;
+  for (const field of Object.keys(response)) {
+    const fieldResponse = response[field];
+    if (fieldResponse.identifier === "email") {
+      prospectEmail = fieldResponse.value as string;
+      break;
+    }
+  }
+  if (!prospectEmail) return null;
+
+  // Determine if the action is an event type redirect
+  if (action.type !== "eventTypeRedirectUrl" || !action.eventTypeId) return null;
+
+  const eventType = await EventTypeRepository.findByIdIncludeHostsAndTeam({ id: action.eventTypeId });
+  if (!eventType || eventType.schedulingType !== SchedulingType.ROUND_ROBIN) return null;
+
+  const eventTypeMetadata = eventType.metadata;
```

Comment:
Shouldn't we parse this?

Reply:
The `eventTypeMetadata` gets passed into `getCRMContactOwnerForRRLeadSkip()` which expects type `Prisma.JsonValue`. It then gets parsed in `getCRMContactOwnerForRRLeadSkip()`